### DISCUSSION TITLE
support for user/driver profile data

### DISCUSSION THREAD

**Discussion by Kosta-Github** [teslamate-org/teslamate#963] [Ideas]
**What would you like to be added**:

I would like to see some support for user/driver profile data, e.g., in order to differentiate between my wife and me as a driver.

**Why is this needed**:

See above.

**Additional context**:

The Tesla API provides `user_id` within the [vehicle data](https://www.teslaapi.io/vehicles/state-and-settings#vehicle-data) response already. But I am unsure, if this is the actual driver profile ID or an ID tied to the tesla account, which would be the same for my wife and myself if we are using the same tesla login...

**Comment by adriankumpf** [comment]
As far as I know the user_id has nothing to do with the driver profile. Presumably it indicates the logged in API user.

**Comment by JakobLichterfeld** [comment]
There is no value in the api for the driver profile from what I know. So it would be only possible to assign drives manually or with some automation, linke drive to location x always user a, drive to location y always user b.

**Comment by oliverpecha** [comment]
Have you found a workaround @Kosta-Github  or does this feature currently have support of some kind?

**Reply by cwanja** [reply]
You'd have to ask Tesla to support this through the API.

**Comment by PalleRaa** [comment]
Now this information have been added in the official Tesla App. I would assume it would also be possible to use this information in Teslamate?

**Reply by oliverpecha** [reply]
it needs to be added to API, not just their official App. Btw, I don't see the new information on my Tesla app yet. What have they added?

**Reply by PalleRaa** [reply]
The app now display which driver profile is driving the car. I assume this would also be available to teslamate, but I might be wrong. ![image](https://github.com/user-attachments/assets/fb8fd7a6-e95b-4910-8f1a-fae963ba86fc)

**Comment by sergio-daniels** [comment]
sounds intressting. I looking as well to see the milage done by other drivers
